query/query.go

```diff
@@ -301,11 +302,14 @@ func (q *RawQuery) validateCalculations() error {
 	if len(q.GroupBy) == 0 {
 		return errCalculationsWithNoGroups
 	}
+	// If the calculation operation is count, the field should be omitted,
+	// in which case the number of events containing the corresponding group key
+	// is returned. All other operations require the field be specified.
 	for _, calc := range q.Calculations {
-		// It's okay to omit the column if the calculation operation is count,
-		// in which case the number of events containing the corresponding group key
-		// is returned. All other operations require the field be specified.
-		if calc.Field == nil && calc.Op != Count {
+		if calc.Field != nil && !calc.Op.RequiresField() {
+			return fmt.Errorf("field specified %v for calculation op %v", *calc.Field, calc.Op)
```

Comment:
nit: maybe just me but an error that reads: `field %v CANNOT be specified for calculation op` makes it easier to understand why an error was thrown

Reply:
Fair enough, will update.